src/server/main_service.cc

```diff
@@ -1234,7 +1234,13 @@ void Service::DispatchCommand(ArgSlice args, SinkReplyBuilder* builder,
 
   dfly_cntx->cid = cid;
 
-  if (!InvokeCmd(cid, args_no_cmd, builder, dfly_cntx)) {
+  // If cmd is an alias, pass it to Invoke so the stats are updated against the alias. By defaults
+  // stats will be updated for cid.name
+  std::optional<std::string_view> orig_cmd_name = std::nullopt;
+  if (registry_.IsAlias(cmd)) {
+    orig_cmd_name = cmd;
+  }
```

Comment:
Another alternative to this lookup would be to return more info from `FindExtended`, maybe a struct with some flags like whether the input was alias or not. 

This current approach seems to be simple, although it comes at the expense of an extra hashmap lookup in the command execution path. 

The alias map would ideally be small though so this lookup shouldn't be too bad. I would be interested to know reviewers thoughts on this though.

Reply:
At the end of the day, swiss tables (flat_hash_map) will use flat storage and this table is relatively small (1-2 elements per command). I don't see how this would have any measurable/observable impact on the hot path.

Reply:
Also I would move this:

```
  std::optional<std::string_view> orig_cmd_name = std::nullopt;
  if (registry_.IsAlias(cmd)) {
    orig_cmd_name = cmd;
  }
```

inside `InvokeCmd` so you don't have to pass it explicitly

Reply:
We lose the `cmd` value which the command was invoked with by the time we get to `InvokeCmd` so I added it here as an argument.

I used `std::optional` as `InvokeCmd` is called in many places so a default keeps changes from being spread out, in some other call sites (eg `MultiCommandSquasher`) there is not even a string to use as no command has been issued.

Reply:
yep but that would be a bug right? So for example, if I use the original command name on a squashing context, we will miss/increment the wrong stat right?

Reply:
I see, I will try to address this.